src/RootHitFile.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    RootHitFile(file::Union{IOStream, AbstractString}; batch_size=10)
+
+represents a `.root.hits` file, which can be iterated or read to yield events, containg data
+on energy depositions, i.e. hits.
+
+A `RootHitFile` is also a Tables.jl compatible row table of events. `batch_size` determines
+the number of events grouped into partitions when `Tables.partitions` is used. To change the
+default of 10, see `DEFAULTS`.
+"""
+struct RootHitFile
+    stream::IOBuffer
+    batch_size::Int
+
+    function RootHitFile(stream::IOStream; batch_size=DEFAULTS.root_hits_batch_size)
+        new(IOBuffer(mmap(stream)), batch_size)
+    end
+end
+
+function RootHitFile(path::AbstractString; batch_size=DEFAULTS.root_hits_batch_size)
+    if occursin(r".root.hits$", path)
+        return RootHitFile(open(path); batch_size=batch_size)
+    else
+        throw(ArgumentError("$path is not a .root.hits file"))
+    end
+end
+
+EventTuple = NamedTuple{
```

Comment:
`EventTuple` is way too generic here, other people may have tuples over different kinds of events. :-)

Also, ideally use `const` when declaring type aliases. One thing to consider: Type aliases for `NamedTuple` can only used in limited ways in dispatch without committing type piracy.

Reply:
If you don't want to dispatch on it, using a `NamedTuple` is perfectly fine, of course. But if you define it this tightly, you probably actually want to define a struct instead.

Reply:
Oh yeah, forgot to make it a const. Thanks :)

And I'll rename it to `RootHitEventTuple` or something like that so as to make clear it's specific to this file format.

And yeah, it's a `NamedTuple` exactly because nothing dispatches on the type.